Ignore tabs in expression

d3dave opened the pull request:
Fixes #87

Constant expression can contain tabs as whitespace just like spaces. This PR tests for this and fixes the expression parser to ignore the tabs just like spaces.

Schamper commented on dissect/cstruct/expression.py:
```diff
@@ -18,7 +18,7 @@ def __init__(self, expression: str):
         self.tokens = []
 
-    def equal(self, token: str, expected: Union[str, str[str]]) -> bool:
+    def equal(self, token: str, expected: Union[str, set[str]]) -> bool:
```
```suggestion
    def equal(self, token: str, expected: str | set[str]) -> bool:
```
May as well use the new syntax here too.

d3dave committed:
Use union type expression in ExpressionTokenizer.equal

Other places already use this syntax so use it to be consistent.

Co-authored-by: Erik Schamper <1254028+Schamper@users.noreply.github.com>

d3dave commented:
@DissectBot agree

Schamper reviewed (changes requested):
If you fix the linter this is good to go :+1:

d3dave committed:
Remove unused typing.Union import

d3dave committed:
Ignore tabs in expression

Constant expression can contain tabs as whitespace just like spaces.
This commit tests for this and fixes the expression parser to ignore
the tabs just like spaces.
